packages/sdk/src/utils/transaction.ts

```diff
@@ -1,40 +1,43 @@
-import { Signer } from 'ethers'
+import { Transaction, WalletClient, createPublicClient, http } from 'viem'
 import { LogLevel, getClient } from '..'
-
-type TransactionResponse = Awaited<ReturnType<Signer['sendTransaction']>>
+import { mainnet, goerli, polygon, arbitrum, optimism } from 'viem/chains'
 
 /**
- * Safe txhash.wait which handles replacements when users speed up the transactio
+ * Safe txhash.wait which handles replacements when users speed up the transaction
  * @param url an URL object
  * @returns A Promise to wait on
  */
 export async function sendTransactionSafely(
+  chainId: number,
   data: any,
-  signer: Signer,
-  setTx: (tx: TransactionResponse) => void,
-  tx?: TransactionResponse
+  signer: WalletClient,
+  setTx: (tx: Transaction['hash']) => void,
+  tx?: Transaction['hash']
 ) {
-  try {
-    let transaction = tx
-    if (!tx) {
-      transaction = await signer.sendTransaction(data)
-      setTx(transaction)
-    }
-    await transaction?.wait()
-    return true
-  } catch (e) {
-    const error = e as any
-    if (
-      error &&
-      error['code'] &&
-      error['code'] === 'TRANSACTION_REPLACED' &&
-      error.replacement
-    ) {
-      setTx(error.replacement)
-      sendTransactionSafely(data, signer, setTx, error.replacement)
-      getClient()?.log(['Transaction replaced', error], LogLevel.Verbose)
-    } else {
-      throw e
-    }
-  }
+  const supportedChains = [mainnet, goerli, polygon, arbitrum, optimism]
```

Comment:
Can we make the supported chains all chains. Just thinking about the friction of every time we add a chain we need to expand this list.

Reply:
So I was running into this issue with Parcel when trying to import allChains like we previously did. Maybe this will be fixed when we switch to rollup

<img width="966" alt="image" src="https://github.com/reservoirprotocol/reservoir-kit/assets/41527174/77c7c08d-8956-4ef9-a849-44c07ea86de6">

Reply:
Hm maybe in the meantime we can just import a bunch of chains one by one. Like you had there but manually import all of them. Can even add a file to rexport them more easily